contracts/zaps/BaseZap.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import {
+    SafeERC20,
+    IERC20
+} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "../../../interfaces/yearn/IYVault.sol";
+
+abstract contract BaseZap is Ownable {
+    using SafeERC20 for IERC20;
+
+    IERC20 public want;
```

Comment:
maybe add allowedTokens to a list to be implemented by extended strat
see this example of protected tokens from baseStrategy:

https://github.com/iearn-finance/yearn-vaults/blob/91e30154df68e0d3c0dbfcdd62ef6512134f8e73/contracts/BaseStrategy.sol#L737

Reply:
In order to add a sweep method?
The difference with a strategy is that the zap contract should never hold any funds, I should add a sweep method to the base contract.

Reply:
Sorry i wasnt clear, this is not for sweep, altho it could be good to add. What i meant is that allowedTokens restricts the tokens you can use with this zap, if we support ERC20s passing the address as argument, that address has to be checked against a well known and restricted list, since it can be any contract. So BaseStrategy implementing allowedTokens for that restriction.

E.g y3crv vault allows ERC20s tokens like = (USDC, DAI, USDT) and also ETH, so zap handles does tokens correctly but i cannot use the zap with cUSDC for example since is not part of the allowed tokens and also the zap may not be able to deal with it correctly. The list is hardcoded per zap so its immutable and can use the same approach as baseStrategy for protected tokens is what i meant.

Yes funds are never meant to be stuck in the zap so could be good to add sweep